core/templates/pages/exploration-editor-page/history-tab/history-tab.component.ts

```diff
@@ -71,7 +71,13 @@ angular.module('oppia').component('historyTab', {
       // Compares the two selected versions and displays the comparison
       // results.
       ctrl.compareSelectedVersions = function() {
-        if (ctrl.selectedVersionsArray.length === 2) {
+        // The selectedVersionsArray may contain empty elements as the
+        // array items are directly manipulated based on indices on change.
+        if (
+          Number.isInteger(ctrl.selectedVersionsArray[0]) &&
+          Number.isInteger(ctrl.selectedVersionsArray[1]) &&
+          ctrl.selectedVersionsArray.length === 2
```

Comment:
```suggestion
          ctrl.selectedVersionsArray.length === 2 &&
          Number.isInteger(ctrl.selectedVersionsArray[0]) &&
          Number.isInteger(ctrl.selectedVersionsArray[1])
```
Lazy evaluation will ensure that the 2nd and 3rd conditions don't get evaluated if the first one fails.

Reply:
done